flutter/lib/src/sentry_user_interaction_widget.dart

```diff
@@ -0,0 +1,288 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+import 'package:sentry/sentry.dart';
+
+import 'widget_click/tapped_widget.dart';
+
+// Adapted from https://github.com/ueman/sentry-dart-tools/blob/8e41418c0f2c62dc88292cf32a4f22e79112b744/sentry_flutter_plus/lib/src/widgets/click_tracker.dart
+
+const _tapDeltaArea = 20 * 20;
+Element? _clickTrackerElement;
+
+class SentryUserInteractionWidget extends StatefulWidget {
+  SentryUserInteractionWidget({
+    Key? key,
+    required this.child,
+    @internal Hub? hub,
+  }) : super(key: key) {
+    _hub = hub ?? HubAdapter();
+  }
+
+  final Widget child;
+
+  late final Hub _hub;
+
+  @override
+  StatefulElement createElement() {
+    final element = super.createElement();
+    _clickTrackerElement = element;
+    return element;
+  }
+
+  @override
+  // ignore: no_logic_in_create_state
+  _SentryUserInteractionWidgetState createState() =>
+      _SentryUserInteractionWidgetState(_hub);
+}
+
+class _SentryUserInteractionWidgetState
+    extends State<SentryUserInteractionWidget> {
+  int? _lastPointerId;
+  Offset? _lastPointerDownLocation;
+  final Hub _hub;
```

Comment:
This shouldn't be here. In a state you can get properties of the widget via `widget.hub`. No need to explicitly pass it to the state.

Reply:
Passing parameters via constructor to the state can even lead to weird behavior, because it gets out of date when the widget itself has changes to what gets passed.